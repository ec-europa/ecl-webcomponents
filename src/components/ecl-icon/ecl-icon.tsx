import { Component, Prop, h } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";

@Component({
  tag: 'ecl-icon',
  styleUrls: {
    ec: './build/styles/ecl-icon-ec.css',
    eu: './build/styles/ecl-icon-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclIcon {
  @Prop() styleClass: string = '';
  @Prop() icon: string = '';
  @Prop() size: string = 'xs';
  @Prop() color: string = '';
  @Prop() titleTag: string = '';
  @Prop({ mutable: true }) path: string;
  @Prop() transform: string = '';
  @Prop({ mutable: true }) theme: string;
  @Prop() sprite: string = '';

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
    
    if (!this.sprite) {
      this.path = getAssetPath(`./build/images/${this.theme}/icons.svg`);
    } else {
      this.path = getAssetPath(`./build/images/${this.theme}/${this.sprite}.svg`);
    }
  }

  getClass(): string {
    const styleClasses = [
      `ecl-icon`,
      `ecl-icon--${this.size}`,
      this.styleClass,
    ];
    if (this.color) {
      styleClasses.push(
        `ecl-icon--${this.color}`
      );
    }
    if (this.transform) {
      styleClasses.push(
        `ecl-icon--${this.transform}`
      );
    }

    return styleClasses.join(' ');
  }

  render() {
    return (
      <svg class={this.getClass()}>
      { this.titleTag ? <title>{this.titleTag}</title> : '' } 
        <use xlinkHref={`${this.path}#${this.icon}`}></use>
      </svg>
    )
  }
}