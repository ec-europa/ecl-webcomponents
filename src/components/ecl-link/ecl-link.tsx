import { Component, h, Prop} from '@stencil/core';

@Component({
  tag: 'ecl-link',
  styleUrls: {
    ec: './build/styles/ecl-link-ec.css',
    eu: './build/styles/ecl-link-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclLink {
  @Prop() theme: string = 'ec';
  @Prop() path: string;
  @Prop() styleClass: string;
  @Prop() variant: string = 'default';
  @Prop() titleAttr: string;
  @Prop() external: boolean = false;

  getClass(): string {
    return [
      `ecl-link`,
      `ecl-link--${this.variant}`,
      this.styleClass
    ].join(' ');
  }

  getExternal() {
    return (
      <ecl-icon 
        icon="external"
        style-class={`ecl-link__icon ecl-link__icon--after sc-ecl-link-${this.theme}`}
        size = "2xs"
      >
      </ecl-icon>
    )  
  }

  render() {
    return (
      <a
        class={this.getClass()}
        href={this.path}
        title={this.titleAttr}
      >
        <slot name="icon-before"></slot>
        <slot></slot>
        <slot name="icon-after">{ this.external ? this.getExternal() : '' }</slot>
      </a>
    );
  }
}
