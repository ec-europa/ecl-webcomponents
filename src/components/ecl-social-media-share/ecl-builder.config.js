const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, './node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  copy: [
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/ec'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/eu'),
      options: { up: true },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-social-media-share-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-social-media-share-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-social-media-share-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-social-media-share-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
