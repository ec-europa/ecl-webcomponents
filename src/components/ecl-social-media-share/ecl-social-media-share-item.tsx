import { Component, h, Prop } from '@stencil/core';

@Component({
  tag: 'ecl-social-media-share-item',
  shadow: false,
})
export class EclSocialMediaShareItem {
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() sharePath: string;
  @Prop() icon: string;

  getClass(): string {
    return [
      `ecl-social-media-share__item`,
      `sc-ecl-social-media-share-${this.theme}`,
      this.styleClass
    ].join(' ');
  }

  render() {
    return (
      <li class={this.getClass()}>
        <ecl-link
          style-class={`ecl-link ecl-link--standalone ecl-link--icon-before ecl-social-media-share__link sc-ecl-social-media-share-${this.theme}`}
          path={this.sharePath}
          icon-position="before"
        >
          <ecl-icon
            slot="icon-before"
            sprite="icons-social-media"
            icon={this.icon}
            size="m"
          ></ecl-icon>
          <slot></slot>
        </ecl-link>
      </li>
    );
  }
}