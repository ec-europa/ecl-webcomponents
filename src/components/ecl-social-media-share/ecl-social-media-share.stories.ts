const getArgs = () => {
  return {
    variant: 'horizontal',
  };
};

const getArgTypes = () => {
  return {
    variant: {
      type: { name: 'select' },
      options: ['horizontal', 'vertical'],
      description: "Social Media Share variant"
    },
  };
};

export default {
  title: 'Components/social media share',
};

const Template = (args) => 
`<ecl-social-media-share
  variant="${args.variant}"
  theme="${args.theme}"
  description="Share this page"
>
  <ecl-social-media-share-item
    theme="${args.theme}"
    icon="facebook-color"
    share-path="http://facebook.com"
  >
    Facebook
  </ecl-social-media-share-item>
  <ecl-social-media-share-item
    theme="${args.theme}"
    icon="twitter-color"
    share-path="http://twitter.com"
  >
    Twitter
  </ecl-social-media-share-item>
  <ecl-social-media-share-item
    theme="${args.theme}"
    icon="instagram-color"
    share-path="http://instagram.com"
  >
    Instagram
  </ecl-social-media-share-item>
  <ecl-social-media-share-item
    theme="${args.theme}"
    icon="linkedin-color"
    share-path="http://linkedin.com"
  >
    Linkedin
  </ecl-social-media-share-item>
  <ecl-social-media-share-item
    theme="${args.theme}"
    icon="telegram-color"
    share-path="http://telegram.com"
  >
    Telegram
  </ecl-social-media-share-item>
  <ecl-popover
    ecl-script
    theme="${args.theme}"
    toggle-label="Other social networks"
    list
    icon="share"
    icon-size="s"
    item-id="popover-example"
  >
    <ecl-popover-item
      theme="${args.theme}"
      path="/example"
      icon="pinterest-color"
      icon-sprite="icons-social-media"
      icon-size="s"
    >
      Pinterest
    </ecl-popover-item>
    <ecl-popover-item
      theme="${args.theme}"
      path="/example"
      icon="mastodon-color"
      icon-sprite="icons-social-media"
      icon-size="s"
    >
      Mastodon
    </ecl-popover-item>
    <ecl-popover-item
      theme="${args.theme}"
      path="/example"
      icon="reddit-color"
      icon-sprite="icons-social-media"
      icon-size="s"
    >
      Reddit
    </ecl-popover-item>
    <ecl-popover-item
      theme="${args.theme}"
      path="/example"
      icon="youtube-color"
      icon-sprite="icons-social-media"
      icon-size="s"
    >
      Youtube
    </ecl-popover-item>
    <ecl-popover-item
      theme="${args.theme}"
      path="/example"
      icon="flickr-color"
      icon-sprite="icons-social-media"
      icon-size="s"
    >
      Flickr
    </ecl-popover-item>
    <ecl-popover-item
      theme="${args.theme}"
      path="/example"
      icon="skype-color"
      icon-sprite="icons-social-media"
      icon-size="s"
    >
      Skype
    </ecl-popover-item>
  </ecl-popover>
</ecl-social-media-share>`;

export const SocialMediaShare = Template.bind({});
SocialMediaShare.storyName = 'default';
SocialMediaShare.args = getArgs();
SocialMediaShare.argTypes = getArgTypes();
