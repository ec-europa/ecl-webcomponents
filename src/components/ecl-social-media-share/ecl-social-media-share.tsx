import { Component, h, Prop } from '@stencil/core';

@Component({
  tag: 'ecl-social-media-share',
  styleUrls: {
    ec: './build/styles/ecl-social-media-share-ec.css',
    eu: './build/styles/ecl-social-media-share-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})
export class EclSocialMediaShare {
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() variant: string = 'horizontal';
  @Prop() description: string;
  @Prop() items: string;
  @Prop() others: string;

  getClass(): string {
    return [
      `ecl-social-media-share`,
      `ecl-social-media-share--${this.variant}`,
      this.styleClass
    ].join(' ');
  }

  componentWillLoad() {
    const patchStyle = document.createElement('style');
    patchStyle.innerHTML = `.ecl-social-media-share__item .ecl-icon { margin-inline-end: 0.5rem; }
                            ecl-icon { display: inline-flex; vertical-align: sub; }`;
    document.body.appendChild(patchStyle);
  }

  render() {
    const itemsArray = this.items ? JSON.parse(this.items) : '';
    const othersObject = this.others ? JSON.parse(this.others) : '';

    return (
      <div 
        class={this.getClass()}
      >
      { this.description ? 
        <p class="ecl-social-media-share__description">{this.description}</p> 
        : '' 
      }
        <ul class="ecl-social-media-share__list">
        {itemsArray.length > 0 ? itemsArray.map((item) => (
          <li class="ecl-social-media-share__item">
            <ecl-link
              style-class="ecl-link ecl-link--standalone ecl-link--icon-before ecl-social-media-share__link"
              path={item.url}
              icon-position="before"
            >
            <ecl-icon
              slot="ecl-icon-before"
              sprite="icons-social-media"
              icon={item.name + `-color`}
              size="m"
            ></ecl-icon>
              {item.name.charAt(0).toUpperCase() + item.name.slice(1)}
            </ecl-link>
          </li>
        )) : ''}
        { othersObject ? 
          <li class="ecl-social-media-share__item ecl-social-media-share__others">
            <ecl-link
              style-class="ecl-link ecl-link--standalone ecl-link--icon-before ecl-social-media-share__link"
              path={othersObject.url}
              icon-position="before"
            >
            <ecl-icon
              slot="ecl-icon-before"
              icon="share"
              size="m"
            ></ecl-icon>
              {othersObject.name}
            </ecl-link>
          </li> : '' }
        </ul>
      </div>
    );
  }
}
