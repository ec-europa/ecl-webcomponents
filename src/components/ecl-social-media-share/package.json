{
  "name": "@ecl/web-components-social-media-share",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.11.0",
  "description": "ECL Social Media Share",
  "main": "ecl-social-media-share.tsx",
  "private": true,
  "scripts": {
    "build:copy": "cross-env ecl-builder copy",
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.10.0",
    "@ecl/resources-social-media-icons": "3.10.0",
    "@ecl/specs-component-icon": "3.10.0",
    "@ecl/vanilla-component-icon": "3.10.0",
    "@ecl/vanilla-component-social-media-share": "3.10.0",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
