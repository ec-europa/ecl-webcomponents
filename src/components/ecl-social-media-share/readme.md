# ecl-social-media-share



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type     | Default     |
| ------------ | ------------- | ----------- | -------- | ----------- |
| `items`      | `items`       |             | `string` | `undefined` |
| `others`     | `others`      |             | `string` | `undefined` |
| `styleClass` | `style-class` |             | `string` | `undefined` |
| `theme`      | `theme`       |             | `string` | `'ec'`      |


## Dependencies

### Depends on

- [ecl-icon](../ecl-icon)

### Graph
```mermaid
graph TD;
  ecl-social-media-share-items --> ecl-icon
  style ecl-social-media-share-items fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
