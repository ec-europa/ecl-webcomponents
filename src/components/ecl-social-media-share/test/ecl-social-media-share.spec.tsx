import { newSpecPage } from '@stencil/core/testing';
import { EclSocialMediaShare } from '../ecl-social-media-share';
import { EclSocialMediaShareItem } from '../ecl-social-media-share-item';
import { EclLink } from '../../ecl-link/ecl-link';
import { EclIcon } from '../../ecl-icon/ecl-icon';
import { EclPopover } from '../../ecl-popover/ecl-popover';

describe('ecl-social-media-share', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclSocialMediaShare, EclSocialMediaShareItem, EclLink, EclIcon, EclPopover],
      html: `<ecl-social-media-share
        variant="horizontal"
        theme="ec"
        description="Share on social media"
      >
        <ecl-social-media-share-item
          icon="facebook-color"
          share-path="http://facebook.com"
        >
          Facebook
        </ecl-social-media-share-item>
        <ecl-social-media-share-item
          icon="twitter-color"
          share-path="http://twitter.com"
        >
          Twitter
        </ecl-social-media-share-item>
        <ecl-social-media-share-item
          icon="instagram-color"
          share-path="http://instagram.com"
        >
          Instagram
        </ecl-social-media-share-item>
        <ecl-social-media-share-item
          icon="linkedin-color"
          share-path="http://linkedin.com"
        >
          Linkedin
        </ecl-social-media-share-item>
        <ecl-social-media-share-item
          icon="telegram-color"
          share-path="http://telegram.com"
        >
          Telegram
        </ecl-social-media-share-item>
        <ecl-popover
          ecl-script
          theme="ec"
          toggle-label="Popover"
          list
          icon="share"
          icon-size="s"
          item-id="popover-example"
        >
          <ecl-popover-item
            path="/example.html"
            icon="pinterest-color"
            icon-sprite="icons-social-media"
            icon-size="s"
          >
            Pinterest
          </ecl-popover-item>
          <ecl-popover-item
            path="/example.html"
            icon="mastodon-color"
            icon-sprite="icons-social-media"
            icon-size="s"
          >
            Mastodon
          </ecl-popover-item>
          <ecl-popover-item
            path="/example.html"
            icon="reddit-color"
            icon-sprite="icons-social-media"
            icon-size="s"
          >
            Reddit
          </ecl-popover-item>
          <ecl-popover-item
            path="/example.html"
            icon="youtube-color"
            icon-sprite="icons-social-media"
            icon-size="s"
          >
            Youtube
          </ecl-popover-item>
          <ecl-popover-item
            path="/example.html"
            icon="flickr-color"
            icon-sprite="icons-social-media"
            icon-size="s"
          >
            Flickr
          </ecl-popover-item>
          <ecl-popover-item
            path="/example.html"
            icon="skype-color"
            icon-sprite="icons-social-media"
            icon-size="s"
          >
            Skype
          </ecl-popover-item>
        </ecl-popover>
      </ecl-social-media-share>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
