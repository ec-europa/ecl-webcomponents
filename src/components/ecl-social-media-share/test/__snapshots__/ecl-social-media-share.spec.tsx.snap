// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-social-media-share renders 1`] = `
<ecl-social-media-share description="Share on social media" theme="ec" variant="horizontal">
  <div class="ecl-social-media-share ecl-social-media-share--horizontal">
    <p class="ecl-social-media-share__description">
      Share on social media
    </p>
    <ul class="ecl-social-media-share__list"></ul>
  </div>
</ecl-social-media-share>
`;
